src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -1,14 +1,61 @@
 package core.basesyntax.service;
 
+import core.basesyntax.InvalidDataException;
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_CHARACTERS = 6;
+    private static final int MIN_AGE = 18;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        validateUser(user);
+        validateLogin(user);
+        validatePassword(user);
+        validateAge(user);
+        return storageDao.add(user);
+    }
+
+    private void validateUser(User user) {
+        if (user == null) {
+            throw new InvalidDataException("User is null");
+        }
+    }
+
+    private void validateLogin(User user) {
+        if (user.getLogin() == null) {
+            throw new InvalidDataException("Invalid login. Null is not a valid value");
+        }
+        if (user.getLogin().length() < MIN_CHARACTERS) {
+            throw new InvalidDataException("This login least than "
```

Comment:
```suggestion
            throw new InvalidDataException("Login shouldn't be less than "
```
fix similar issue in all places